mixnet/node/src/lib.rs

```diff
@@ -178,14 +197,17 @@ impl MixnetNode {
         .await
     }
 
-    async fn forward(
-        pool: &ConnectionPool,
-        body: Body,
-        to: NymNodeRoutingAddress,
-    ) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-        let addr = SocketAddr::try_from(to)?;
-        body.write(&mut *pool.get_or_init(&addr).await?.lock().await)
-            .await?;
+    async fn forward(pool: &ConnectionPool, body: Body, to: NymNodeRoutingAddress) -> Result<()> {
+        let addr = SocketAddr::from(to);
+        body.write(
+            &mut *pool
+                .get_or_init(&addr)
+                .await
+                .map_err(ProtocolError::IO)?
+                .lock()
+                .await,
```

Comment:
I guess we can make `poo.get_or_init` return `ProtocolError::IO`, so that we don't need to do `map_err()` whenever we use `get_or_init`.

Reply:
`💩.get_or_init`